scribe/lib/scribe/ai/presentation/widgets/ai_scribe_bar.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:flutter/material.dart';
+import 'package:core/presentation/resources/image_paths.dart';
+import 'package:core/presentation/views/button/tmail_button_widget.dart';
+import 'package:scribe/scribe/ai/localizations/scribe_localizations.dart';
+import 'package:scribe/scribe/ai/presentation/styles/ai_scribe_styles.dart';
+
+typedef OnCustomPromptCallback = void Function(String customPrompt);
+
+class AIScribeBar extends StatefulWidget {
+  final OnCustomPromptCallback onCustomPrompt;
+  final ImagePaths imagePaths;
+
+  const AIScribeBar({
+    super.key,
+    required this.onCustomPrompt,
+    required this.imagePaths,
+  });
+
+  @override
+  State<AIScribeBar> createState() => _AIScribeBarState();
+}
+
+class _AIScribeBarState extends State<AIScribeBar> {
+  final TextEditingController _controller = TextEditingController();
+  bool _isButtonEnabled = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller.addListener(_onTextChanged);
+  }
+
+  @override
+  void dispose() {
+    _controller.removeListener(_onTextChanged);
+    _controller.dispose();
+    super.dispose();
+  }
+
+  void _onTextChanged() {
+    setState(() {
+      _isButtonEnabled = _controller.text.trim().isNotEmpty;
+    });
```

Comment:
If only a small widget is being changed, use `ValueNotifier` instead of `setState`.